packages/ui5-test-writer/src/index.ts

```diff
@@ -0,0 +1,202 @@
+        target.name in SupportedPageTypes &&
+        target?.id &&
+        target?.options?.settings?.entitySet
+    ) {
+        return {
+            appPath,
+            appID,
+            targetKey,
+            componentID: target.id,
+            entitySet: target.options.settings.entitySet,
+            template: SupportedPageTypes[target.name],
+            isStartup: false
+        };
+    } else {
+        return undefined;
+    }
+}
+/**
+ * Create the configuration object from the app descriptor.
+ *
+ * @param manifest - the app descriptor of the target app
+ * @returns OPA test configuration object
+ */
+function createConfig(manifest: Manifest): FEV4OPAConfig {
+    // General application info
+    const appID = manifest['sap.app'].id;
+    const appPath = appID.split('.').join('/');
+
+    const config: FEV4OPAConfig = {
+        appID,
+        appPath,
+        pages: []
+    };
+
+    // Identify startup targets from the routes
+    const appRoutes = (manifest['sap.ui5']?.routing?.routes ?? []) as any[];
+    // Find the route with an empty pattern (except for the trailing query part)
+    const startupRoute = appRoutes.find((route: { pattern: string }) => {
+        return route.pattern.replace(':?query:', '') === '';
+    });
+    let startupTargets = startupRoute?.target ?? [];
+    if (!Array.isArray(startupTargets)) {
+        startupTargets = [startupTargets];
+    }
+
+    // Create page configurations in supported cases
+    const appTargets = manifest['sap.ui5']?.routing?.targets;
+    for (const targetKey in appTargets) {
+        const pageConfig = createPageConfig(manifest, targetKey);
+        if (pageConfig) {
+            pageConfig.isStartup = startupTargets.includes(targetKey);
+            config.pages.push(pageConfig);
+        }
+    }
+
+    return config;
+}
+
+/**
+ * Writes a page object in a mem-fs-editor.
+ *
+ * @param pageConfig - the page configuration object
+ * @param rootTemplateDirPath - template root directory
+ * @param testOutDirPath - output test directory (.../webapp/test)
+ * @param fs - a reference to a mem-fs editor
+ */
+function writePageObject(
+    pageConfig: FEV4OPAPageConfig,
+    rootTemplateDirPath: string,
+    testOutDirPath: string,
+    fs: Editor
+) {
+    fs.copyTpl(
+        join(rootTemplateDirPath, `integration/pages/${pageConfig.template}.js`),
+        join(testOutDirPath, `integration/pages/${pageConfig.targetKey}.js`),
+        pageConfig,
+        undefined,
+        {
+            globOptions: { dot: true }
+        }
+    );
+}
+
+/**
+ * Generate OPA test files for a Fiori elements for OData V4 application.
+ * Note: this can potentially overwrite existing files in the webapp/test folder.
+ *
+ * @param basePath - the absolute target path where the application will be generated
+ * @param fs - an optional reference to a mem-fs editor
+ * @returns Reference to a mem-fs-editor
+ */
+export function generateOPAFiles(basePath: string, fs?: Editor): Editor {
```

Comment:
Please add an additional parameter (as the 2nd one) to allow providing configurations, so that we can make it more flexible in the future. For now I would just expect `opaConfig: { scriptName?: string }` allowing to provide the name of a script that should be added to run the opa tests.

Reply:
`generateOPAFiles` and `generatePageObjectFile` have 2 different usages.
- `generateOPAFiles` generates all test files for a given project, i.e. the content of the 'test' folder. It should be called only once for a given project (files are overwritten), e.g. from the Fiori tools app generator.
- `generatePageObjectFile` generates an additional page object file in `test/integration/pages`. It can be called each time a new page is added in the app, e.g. using the wizard from Fiori tools.